### PR TITLE
[OKX] add okx futures tests

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/837
requires https://github.com/Drakkar-Software/OctoBot-Tentacles/pull/847

### CODE REVIEW

File: exchanges_tests/abstract_authenticated_future_exchange_tester.py [hunk 1]
```diff
@@ -38,19 +42,103 @@ async def test_get_empty_linear_and_inverse_positions(self):
...
+        if not self.SUPPORTS_SET_LEVERAGE:
+            return
+        new_leverage = origin_leverage + 1
+        await self.set_leverage(new_leverage)
+        await self._check_margin_type_and_leverage(origin_margin_type, new_leverage)    # did not change margin type
+        # change leverage back to origin value
+        await self.set_leverage(origin_leverage)
+        await self._check_margin_type_and_leverage(origin_margin_type, origin_leverage)  # did not change margin type
+
+    async def test_get_and_set_margin_type(self):
```
Comment (Herklos) [inline exchanges_tests/abstract_authenticated_future_exchange_tester.py:81]: 👍

File: exchanges_tests/test_okx_futures.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#  License along with OctoBot. If not, see <https://www.gnu.org/licenses/>.
+import pytest
+
+from exchanges_tests import abstract_authenticated_future_exchange_tester
+
+# All test coroutines will be treated as marked.
+pytestmark = pytest.mark.asyncio
+
+
+class TestOKXFuturesAuthenticatedExchange(
```
Comment (Herklos) [inline exchanges_tests/test_okx_futures.py:24]: 👍

**Review by Herklos — APPROVED** [review]
Great test framework improvements!

**Comment by GuillaumeDSM** [comment]
merged with https://github.com/Drakkar-Software/OctoBot/pull/2251

**Pull request closed without merging**